packages/material-ui/src/Button/Button.js

```diff
@@ -84,7 +84,7 @@ const ButtonRoot = experimentalStyled(ButtonBase, {
         duration: theme.transitions.duration.short,
       },
     ),
-    '&:hover': {
+    '&:hover': !styleProps.disabled && {
```

Comment:
Looks good, I would maybe write it like this:

```
...(!styleProps.disabled && {
  '&:hover': {...}
}),

```

So that we don't end up with `'&:hover': false`, but even that should not create any issues I think.

Reply:
Personally, I'm having a hard time parsing the code you wrote. And it's creating an additional object. 

> So that we don't end up with '&:hover': false, but even that should not create any issues I think.

Yeah, let's try this out then.

Reply:
This is prone to mistakes. For example, we also have '&:hover' on line 229 https://github.com/mui-org/material-ui/blob/8e0eb6d678243e6ffc73145ba3f8a1d30b78bc57/packages/material-ui/src/Button/Button.js#L229

In addition, what other components should incorporate this logic?